test/lib/course_planner/offered_courses_test.exs

```diff
@@ -73,4 +73,135 @@ defmodule CoursePlanner.OfferedCoursesTest do
       assert List.first(courses).id == user_course.id
     end
   end
+
+  describe "get_offered_courses_with_not_filled_class_attendances" do
+    test "when there is no offered_course" do
+      assert [] == OfferedCourses.get_offered_courses_with_not_filled_class_attendances()
+    end
+
+    test "when there is no class for an offered_course" do
+      insert(:offered_course)
+      assert [] == OfferedCourses.get_offered_courses_with_not_filled_class_attendances()
+    end
+
+    test "when there is no attendance for class" do
+      students = insert_list(3, :student)
+      teacher = insert(:teacher)
+      class = insert(:class, date: Timex.now())
+      insert(:offered_course, classes: [class], students: students, teachers: [teacher])
+
+      requested_current_date =  Timex.shift(Timex.now(), days: 2)
+      assert [] == OfferedCourses.get_offered_courses_with_not_filled_class_attendances(requested_current_date)
+    end
+
+    test "when the class is in future" do
+      students = insert_list(3, :student)
+      teacher = insert(:teacher)
+      class = insert(:class, date: Timex.shift(Timex.now(), days: 2))
+      Enum.each(students, fn(student) ->
+        insert(:attendance, student: student, class: class, attendance_type: "Not filled")
+      end)
+      insert(:offered_course, classes: [class], students: students, teachers: [teacher])
+
+      assert [] == OfferedCourses.get_offered_courses_with_not_filled_class_attendances()
+    end
+
+    test "when all attendances of the class are filled for an offered_course" do
+     students = insert_list(3, :student)
+     teacher = insert(:teacher)
+     class = insert(:class, date: Timex.now())
+     Enum.each(students, fn(student) ->
+       insert(:attendance, student: student, class: class, attendance_type: "Present")
+     end)
+     insert(:offered_course, classes: [class], students: students, teachers: [teacher])
+
+     requested_current_date =  Timex.shift(Timex.now(), days: 2)
+     assert [] == OfferedCourses.get_offered_courses_with_not_filled_class_attendances(requested_current_date)
+    end
+
+    test "when there is missing attendances for one class of an offered_course" do
+     students = insert_list(3, :student)
+     teacher = insert(:teacher)
+     class = insert(:class, date: Timex.now())
+     Enum.each(students, fn(student) ->
+       insert(:attendance, student: student, class: class, attendance_type: "Not filled")
+     end)
+     offered_course =
+       insert(:offered_course, classes: [class], students: students, teachers: [teacher])
+
+     requested_current_date =  Timex.shift(Timex.now(), days: 2)
```

Comment:
what do you think of using `requested_current_date` as the actual current date, but create the class two days in the past? it is more understandable in my opinion